docs/authoring-recipes/automate-breaking-changes.md

```diff
@@ -0,0 +1,255 @@
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+import static org.openrewrite.gradle.Assertions.buildGradle;
+
+class UpgradeToSpringBoot3Test implements RewriteTest {
+
+  @Override
+  public void defaults(RecipeSpec spec) {
+      // Note: You normally wouldn't have the snippet of `dependsOn`. 
+      //       This was done simply for having an example that works.
+      spec.recipeFromResources("com.acme.UpgradeToSpringBoot3")
+        .parser(JavaParser.fromJavaVersion().dependsOn(
+          """
+          package com.example.springboot2.starter;
+
+          public class MyConfig {
+              void setTimeout(int t) {}
+              int getTimeout() { return 10; }
+          }
+          """));
+  }
+
+  @Test
+  void replacesStringEquals() {
+    rewriteRun(
+      // Note: This test will actually fail due to the fact that `inhouse-starter` doesn't exist 
+      //       so `2.0.x` won't figure out the latest minor version as there is none. 
+      //       For a real library, this would work fine, though.
+      //language=groovy
+      buildGradle(
+        """
+            plugins {
+                id 'java-library'
+            }
+            repositories {
+                mavenCentral()
+            }
+            dependencies {
+                implementation("com.acme:inhouse-starter:1.0.0")
+            }
+            """,
+        """
+            plugins {
+                id 'java-library'
+            }
+            repositories {
+                mavenCentral()
+            }
+            dependencies {
+                implementation("com.acme:inhouse-starter:2.0.x")
+            }
+            """
+      ),
+      //language=java
+      java(
+        """
+          import com.example.springboot2.starter.foo;
+          import com.example.springboot2.starter.MyConfig;
+          
+          public class MyApp {
+              public static void main(String[] args) {
+                  MyConfig config = new MyConfig();
+                  config.setTimeout(1000);
+                  System.out.println(config.getTimeout());
+              }
+          }
+          """,
+        """
+          import com.example.springboot3.starter.foo;
+          import com.example.springboot3.starter.MyNewConfig;
+
+          public class MyApp {
+              public static void main(String[] args) {
+                  MyNewConfig config = new MyNewConfig();
+                  config.setTimeout(1000);
+                  System.out.println(config.getTimeout());
+              }
+          }
+          """
+      )
+    );
+  }
+}
+```
+</details>
+
+## Step 5: Package/distribute the recipe
+
+Once you've finished creating/testing your recipe, the next thing you'll need to do is figure out how you want to distribute it to others.
+
+TODO: What should we say here?
```

Comment:
Here we can say that folks can package and distribute this recipe as any other library to Maven Central, or an in house Artifactory/Nexus repository.

We recommend creating a separate library to contain the recipes, and not merge these into any existing library. That should make it easier to independently develop and release recipes for upcoming versions, and in the future chain together migration recipes for new versions.